### PR TITLE
Add customizable night vision strength feature

### PR DESCRIPTION
**Pull request by Manchick0**
Introduced a `nightVision` subcommand to adjust night vision strength with options for `full` (= 100), `off` (= 0), and any other value between 0 and 100. Integrated the setting into the configuration menu and game rendering, with default strength set to 100. Updated localization to include feedback for strength changes.

**Explanation**: Night Vision isn't an effect you'd want to have applied all the time. With this feature, you can easily toggle it on and off, or even just slightly reduce its strength, all without interacting with your accessories / booster cookie.

### CODE REVIEW

File: src/main/resources/skyblocker.mixins.json
Comment (kevinthegreat1) [inline src/main/resources/skyblocker.mixins.json:1]: Please don't reformat. This is not even the right format.

File: src/main/java/de/hysky/skyblocker/mixins/GameRendererMixin.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import org.spongepowered.asm.mixin.injection.At;
+
+@Mixin(GameRenderer.class)
+public class GameRendererMixin {
+
+	@ModifyReturnValue(method = "getNightVisionStrength", at = @At("RETURN"))
+	private static float onGetNightVisionStrength(float original) {
+		if (original == 1.0F && Utils.isOnSkyblock()) {
+			var strength = SkyblockerConfigManager.get().uiAndVisuals.nightVisionStrength;
+			return Math.min(Math.max(strength == 0F ? 0F : strength / 100.0F, 0), 100);
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/mixins/GameRendererMixin.java:17]:
```suggestion
			return Math.clamp(strength / 100.0F, 0, 100);
```

File: src/main/java/de/hysky/skyblocker/config/configs/UIAndVisualsConfig.java [hunk 2]
```diff
@@ -6,10 +6,15 @@
 import net.minecraft.util.Formatting;
 
 import java.awt.*;
+import java.io.Serial;
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/config/configs/UIAndVisualsConfig.java:9]: ?

File: src/main/java/de/hysky/skyblocker/config/categories/UIAndVisualsCategory.java [hunk 3]
```diff
@@ -22,7 +22,13 @@ public class UIAndVisualsCategory {
...
                 .name(Text.translatable("skyblocker.config.uiAndVisuals"))
-
+				.option(Option.<Integer>createBuilder()
+						.name(Text.translatable("skyblocker.config.uiAndVisuals.nightVisionStrength"))
+						.binding(defaults.uiAndVisuals.nightVisionStrength,
+								() -> config.uiAndVisuals.nightVisionStrength,
+								newValue -> config.uiAndVisuals.nightVisionStrength = newValue)
+						.controller(opt -> IntegerSliderControllerBuilder.create(opt).range(0, 100).step(1))
+						.build())
                 //Ungrouped Options
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/config/categories/UIAndVisualsCategory.java:32]: After this comment please.

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]

File: src/main/resources/skyblocker.mixins.json
Comment (Manchick0) [reply src/main/resources/skyblocker.mixins.json:1]: It seems like the Minecraft Development Plugin did it on its own. I appologize.

Comment (Manchick0) [reply src/main/resources/skyblocker.mixins.json:1]: And isn't the format right?

````json
{
    "required": true,
    "package": "de.hysky.skyblocker.mixins",
    "plugin": "de.hysky.skyblocker.compatibility.MixinPlugin",
    "compatibilityLevel": "JAVA_21",
    "client": [
        "..."
    ],
    "injectors": {
        "defaultRequire": 1
  }
}

```

File: src/main/java/de/hysky/skyblocker/config/categories/UIAndVisualsCategory.java [hunk 4]
```diff
@@ -27,9 +27,16 @@
 public class UIAndVisualsCategory {
     public static ConfigCategory create(SkyblockerConfig defaults, SkyblockerConfig config) {
         return ConfigCategory.createBuilder()
-                .name(Text.translatable("skyblocker.config.uiAndVisuals"))
-
+				.name(Text.translatable("skyblocker.config.uiAndVisuals"))
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/config/categories/UIAndVisualsCategory.java:30]:
```suggestion
				.name(Text.translatable("skyblocker.config.uiAndVisuals"))

```

File: src/main/java/de/hysky/skyblocker/config/configs/UIAndVisualsConfig.java [hunk 5]
```diff
@@ -11,6 +11,10 @@
 import java.util.List;
 
 public class UIAndVisualsConfig {
+
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/config/configs/UIAndVisualsConfig.java:14]:
```suggestion
```

**Review by kevinthegreat1 — APPROVED** [review]

**Pull request merged**